l10n_fr_siren_lookup/models/res_partner.py

```diff
@@ -0,0 +1,291 @@
+                    _(
+                        "Failure in the request on data.opendatasoft.com "
+                        "to create or update partner from SIREN or SIRET. "
+                        "Technical error: %s."
+                    )
+                    % e
+                )
+        return False
+
+    @api.model
+    def _opendatasoft_parse_record(
+        self, raw_record, exclude_dead=False, vat_vies_query=True
+    ):
+        res = False
+        if raw_record and isinstance(raw_record, dict):
+            if exclude_dead and raw_record.get("datefermetureunitelegale"):
+                return res
+            if exclude_dead and raw_record.get("datefermetureetablissement"):
+                return res
+            res = {
+                "name": raw_record.get("denominationunitelegale")
+                or raw_record.get("l1_adressage_unitelegale"),
+                "street": raw_record.get("adresseetablissement"),
+                "zip": raw_record.get("codepostaletablissement"),
+                "city": raw_record.get("libellecommuneetablissement"),
+                "siren": raw_record.get("siren") and str(raw_record["siren"]) or False,
+                "nic": raw_record.get("nic"),
+            }
+            if raw_record.get("codedepartementetablissement"):
+                dpt_code = raw_record["codedepartementetablissement"]
+                res["country_id"] = self._opendatasoft_dpt2country(dpt_code)
+            # set lang to French if installed
+            fr_lang = self.env["res.lang"].search([("code", "=", "fr_FR")])
+            if fr_lang:
+                res["lang"] = "fr_FR"
+            if res.get("siren") and vat_vies_query:
+                vat = self._siren2vat_vies(res["siren"])
+                if vat is not None:
+                    res["vat"] = vat
+        return res
+
+    @api.model
+    def _opendatasoft_dpt2country(self, dpt_code):
+        domtom2xmlid = {
+            "971": "gp",
+            "972": "mq",
+            "973": "gf",
+            "974": "re",
+            "975": "pm",  # Saint Pierre and Miquelon
+            "976": "yt",  # Mayotte
+            "977": "bl",  # Saint-Barthélemy
+            "978": "mf",  # Saint-Martin
+            "986": "wf",  # Wallis-et-Futuna
+            "987": "pf",  # Polynésie française
+            "988": "nc",  # Nouvelle calédonie
+        }
+        country_id = False
+        if dpt_code and len(dpt_code) == 2:
+            country_id = self.env.ref("base.fr").id
+        elif dpt_code in domtom2xmlid:
+            country_xmlid = "base.%s" % domtom2xmlid[dpt_code]
+            country_id = self.env.ref(country_xmlid).id
+        return country_id
+
+    @api.model
+    def _siren2vat_vies(self, siren, raise_if_fail=False):
+        vat = "FR%s" % siren_to_vat(siren)
+        logger.info("VIES check of VAT %s" % vat)
+        vies_res = False
+        res = False
+        try:
+            vies_res = check_vies(vat, timeout=5)
```

Comment:
```suggestion
            vies_res = check_vies(vat)
```
Timeout is not supported in check_vies for python-stdnum version 1.8 used by Odoo v14.0

Reply:
arf, you should not use these old stdnum versions that don't support the new format for Northern Ireland VAT numbers. Maybe we could check the stdnum version in the code and add the argument when stdnum is 1.9+.

Reply:
Good idea to make the command dependant of stdnum version.
I guess we cannot assume that anyone using this module would update stdnum from Odoo required version (nor may care about Northern Ireland VAT :))

Reply:
Poor Northern Ireland people! Many companies prefer to stick to an old outdated stdnum version because they want to stick to what is written in requirements.txt of Odoo and, as a consequence, they can't sell to Northern Ireland any more. I guess the problem is not just with odoo... I would be interested to know how many companies refuse to sell to Northern Ireland for software reasons! No surprise their shops are empty. And, in Odoo v15, requirements.txt is still on stdnum 1.8, cf https://github.com/odoo/odoo/blob/15.0/requirements.txt#L49 This is not serious, so just don't follow requirements.txt of Odoo.